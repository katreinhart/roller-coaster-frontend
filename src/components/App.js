import React from 'react'
import RollerCoasterTable from './RollerCoasterTable'


function App() {
  return (
    <div className='container'>
      <div className='jumbotron'>
        <div className='display-4 text-center'>ðŸŽ¢ Rollercoasters List! ðŸŽ¢</div>
      </div>
      <div className='container'>
        <div className='row'>
          <RollerCoasterTable />
        </div>
      </div>
    </div>
  )
}

export default App
