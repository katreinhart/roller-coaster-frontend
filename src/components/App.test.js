import React from 'react'
import { shallow } from 'enzyme'

import App from './App'

describe('App', () => {
  const app = shallow(<App />)

  it('renders without crashing', () => {
    expect(app).toMatchSnapshot()
  })

  it('contains a jumbotron with text Rollercoasters List', () => {
    expect(app.find('.jumbotron').exists()).toBe(true)
    expect(app.find('.jumbotron').text()).toContain('Rollercoasters List')
  })

  it ('renders a RollerCoasterTable component', () => {
    expect(app.find('RollerCoasterTable').exists()).toBe(true)
  })
})