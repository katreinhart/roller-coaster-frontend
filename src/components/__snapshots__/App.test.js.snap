// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="jumbotron"
>
          <div
                    className="display-4 text-center"
          >
                    ðŸŽ¢ Rollercoasters List! ðŸŽ¢
          </div>
</div>,
        <div
          className="container"
>
          <div
                    className="row"
          >
                    <RollerCoasterTable />
          </div>
</div>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="display-4 text-center"
>
            ðŸŽ¢ Rollercoasters List! ðŸŽ¢
</div>,
          "className": "jumbotron",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ðŸŽ¢ Rollercoasters List! ðŸŽ¢",
            "className": "display-4 text-center",
          },
          "ref": null,
          "rendered": "ðŸŽ¢ Rollercoasters List! ðŸŽ¢",
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
>
            <RollerCoasterTable />
</div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <RollerCoasterTable />,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="jumbotron"
>
            <div
                        className="display-4 text-center"
            >
                        ðŸŽ¢ Rollercoasters List! ðŸŽ¢
            </div>
</div>,
          <div
            className="container"
>
            <div
                        className="row"
            >
                        <RollerCoasterTable />
            </div>
</div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="display-4 text-center"
>
              ðŸŽ¢ Rollercoasters List! ðŸŽ¢
</div>,
            "className": "jumbotron",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ðŸŽ¢ Rollercoasters List! ðŸŽ¢",
              "className": "display-4 text-center",
            },
            "ref": null,
            "rendered": "ðŸŽ¢ Rollercoasters List! ðŸŽ¢",
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
>
              <RollerCoasterTable />
</div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <RollerCoasterTable />,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
